qemu/tests/virt_firmware_check_phys_bits.py

```diff
@@ -0,0 +1,96 @@
+       make sure it equals to the set value or equals to the maximum
+       number which host supporting
+    3) if phys-bits number is greater than 46, check the limitation
+       message in virt firmware log
+       limitation message: limit PhysBits to 46 (avoid 5-level paging)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    phys_bits_grep_cmd = params["phys_bits_grep_cmd"]
+    host_phys_bits = process.getoutput(phys_bits_grep_cmd, shell=True).strip()
+    if not host_phys_bits.isdigit():
+        test.error("Failed to get host phys-bits, the actual output is '%s'"
+                   % host_phys_bits)
+    host_phys_bits_limit = params["host_phys_bits_limit"]
+    params["cpu_model_flags"] %= host_phys_bits_limit
+    err_msg = params.get("err_msg")
+    ignored_err_msg = params.get("ignored_err_msg")
+    try:
+        error_context.context("Start the vm with host-phys-bits-limit=%s."
+                              % host_phys_bits_limit, test.log.info)
+        vm_name = params["main_vm"]
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        vm.verify_alive()
+    except virt_vm.VMCreateError as e:
+        if err_msg:
+            if err_msg not in str(e):
+                test.fail("Boot a vm with invalid phys-bits '%s', "
+                          "the error message is not the expected value "
+                          "'%s'. The actual output is '%s'."
+                          % (host_phys_bits_limit, err_msg, str(e)))
+        elif ignored_err_msg:
+            if not re.search(ignored_err_msg, str(e), re.S | re.I):
+                test.fail("Boot a vm with phys-bits '%s', the ignored "
+                          "error message is not the expected value "
+                          "'%s'. The actual output is '%s'."
+                          % (host_phys_bits_limit, ignored_err_msg, str(e)))
+        else:
+            test.error("Failed to create a vm, the error is '%s'" % str(e))
+    else:
+        error_context.context("Check the phys-bits in guest.", test.log.info)
+        session = vm.wait_for_login()
+        guest_phys_bits = session.cmd_output(phys_bits_grep_cmd).strip()
+        sev_status = sev_es_status = False
+        if params.get("check_sev_cmd"):
+            output = process.getoutput(params["check_sev_cmd"], shell=True)
+            if output in params["enabled_status"]:
+                sev_status = True
+        if params.get("check_sev_es_cmd"):
+            output = process.getoutput(params["check_sev_es_cmd"], shell=True)
+            if output in params["enabled_status"]:
+                sev_es_status = True
+        if sev_status or sev_es_status:
+            host_phys_bits = int(host_phys_bits) + \
+                    int(params["host_memory_encryption_bits"])
+        expected_phys_bits = min(str(host_phys_bits), host_phys_bits_limit)
```

Comment:
do we want to use a string compare? or a digital compare? it looks to me host phy bits more like an int

Reply:
The phys-bits value is a string in the test, except for one place (line 82, need to sum). So it will be converted to string in line 84.
And for function min(), the parameters to compare in size can be comparable data types such as numbers, strings, lists, tuples, etc. So I think it should be OK here.